internal/kms/pubsub.go

```diff
@@ -327,6 +357,43 @@ func (s *pubSubService) getEncryptionKeysLocally(
 	return blocks, nil
 }
 
+func (s *pubSubService) doesIdentityHaveDocPermission(
+	ctx context.Context,
+	actorIdentity string,
+	entBlock *coreblock.Encryption,
+) (bool, error) {
+	if !s.acp.HasValue() {
+		return true, nil
+	}
+
+	docID := string(entBlock.DocID)
+	collection, err := s.colRetriever.RetrieveCollectionFromDocID(ctx, docID)
```

Comment:
todo: It looks like the rest of this function from this line down is just a duplication of `permission.CheckAccessOfDocOnCollectionWithACP` - can we use that function instead?

Reply:
+1

Reply:
done

Reply:
`permission.CheckAccessOfDocOnCollectionWithACP` takes `immutable.Option[Identity]`. What I have in this functoin is just DID string. So I passed `immutable.Some(identity.Identity{DID: actorIdentity})` to the function. Not sure if it's 100% correct.